salt/roster/dir.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+'''
+Read in the rosters in a directory using the renderer system.
+'''
+from __future__ import absolute_import
+
+# Import python libs
+import os
+from fnmatch import fnmatch
+import re
+import copy
+import logging
+
+import salt.loader
+import salt.config
+from salt.template import compile_template
+
+from salt.roster.flat import HAS_RANGE, _convert_range_to_list
```

Comment:
would you mind moving these to `salt.utils.roster` so that they can just be used by ever roster?

Reply:
still todo

Reply:
@gtmanfred This is done -- please let me know if what I did is what you had in mind.